interface.go

```diff
@@ -344,6 +344,15 @@ type ConnectionState struct {
 	// If datagram support was negotiated, datagrams can be sent and received using the
 	// SendDatagram and ReceiveDatagram methods on the Connection.
 	SupportsDatagrams bool
+	// MaxDatagramSize specifies how big a datagram can be sent using
+	// SendDatagram. Datagrams bigger than MaxDatagramSize are silently dropped.
+	//
+	// Note: MaxDatagramSize can grow or shrink at any time. Users should call
+	// ConnectionState every once in a while to get up to date MaxDatagramSize.
+	//
+	// MaxDatagramSize is zero if datagrams are not supported on this
+	// connection.
+	MaxDatagramSize int
```

Comment:
uint16?

Reply:
I don't mind making it `uint16`, but I don't see what improvement this would have. At worst I believe this would, in some cases, cause the users to perform additional conversion in their code, as dealing with sizes in Go conventionally is done in ints.